packages/core/src/types/deploy.ts

```diff
@@ -210,18 +213,20 @@ export interface DeploymentParameters {
 }
 
 /**
- * An enumeration of the different types of deployment strategies.
+ * The different types of deployment strategies.
  *
  * @beta
  */
-export enum DeploymentStrategyType {
-  /**
-   * The basic deployment strategy.
-   */
-  BASIC = "basic",
+export type DeploymentStrategyType = Create2Strategy | BasicStrategy;
```

Comment:
I don't understand why we are exposing this. We were previously initializing the strategy based on an enum. We discussed changing it to a key of `StrategyConfig` instead, as shown [here](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMpinSBzAngYQHsQZhtkBvAKGVuQCM4BnYBALkoF8AaGuhKBCwQATB2p1JyJnAA2YDkwyhsAbj61O6zlSowAriARhgxZABMIAB1mFcACgCUHAG6Fg59QaMmzlm3YAPOiYOLgAKsgQAB6QIOZMyADWELiEMGgYwnhEJGTIALzIAESMLAjFAHz2SqEQeBwh2RHc0llhuaTYje31BMRdANpNYeEAus7Ibh5ehsamIBbWtrjBvXiRMXEJyanpmXU5A-lFpcysVTXruAD8PYctbQ+dZHcHzS-Yw9fjk9PmlF0Oio-hWTnUoLs9jO5WKrQonEcEOWUOKAiEkBEcMo0jkChKzHMxWQiNUQA)

I don't get why we are now exposing the strategies, this brought about more changes than needed.

Reply:
I had thought it was the design we had coming out of the strategies config document discussion: https://www.notion.so/nomicfoundation/Create2-Configuration-795c08cfa9244f03b4bb9f6e4de5ba07?pvs=4

I may have misunderstood the decision made there.

Why pass a strategy object rather than `"basic" | "create2"` into core's `deploy` function? I was trying to avoid a breaking change on the core api when Strategies proper are rolled out. With a full strategy API I expect that we will pass a strategy object rather than a strategy factory + config that are unified in core.

In my mind the three options are:

1. Pass initialized strategies into `deploy`, but have to expose two core strategies now
2. Pass config and an enumeration now, unifying them within core. Then later support passing a strategy factory in addition as well as the enumeration. Basically always support the current API in addition to the new API of passing a strategy factory along with the strategy config.
3. Pass config and an enumeration now. Breaking change on the API with the full strategy API. 

Is there a better approach? Or is something like 2 what you would prefer to see?